### DISCUSSION TITLE
not working

### DISCUSSION THREAD

**Discussion by Bcawesomer** [HarukaMa/starfield-xgp-import#4] [Q&A]
open the exe drop the file in it closes nothing happens but the exe in the save folder click a key it closes.

**Comment by HarukaMa** [comment]
Try running the program in a cmd window instead to see its output (if there is any).

**Comment by Bcawesomer** [comment]
![Desktop Screenshot 2023 09 06 - 16 19 55 85](https://github.com/HarukaMa/starfield-xgp-import/assets/130524405/1b8bdf17-af38-419a-8ad5-7d0ba64f17e1)
am i doing something wrong?

**Comment by HarukaMa** [comment]
drag the exe file into the cmd window, add a space ~~and a double quote~~ after that, drag the sfs file into the cmd window ~~, then add another double quote~~. the whole command line should look like 
```
<... path ...>\sfsconvert.exe "<... path ...>\Save74_...sfs"
```

**Comment by Bcawesomer** [comment]
![Desktop Screenshot 2023 09 06 - 16 29 12 18](https://github.com/HarukaMa/starfield-xgp-import/assets/130524405/c9c2c006-3f7a-41e5-90d0-c032948038c9)

**Comment by scwiterx** [comment]
Here I managed to convert, but the game does not recognize the save.

**Comment by HarukaMa** [comment]
@Bcawesomer didn't know you don't need to manually input the double quotes anymore. Please try again with just manually inputting a space.

@scwiterx The importer completed without any error, and the game is not showing the save file when you try tho load the game? Please first make sure you are importing the save file with the game fully closed and wait for a short moment after closing the game. Otherwise I might need the current container and the backup one to further diagnose it.

**Comment by Bcawesomer** [comment]
i just found that out thanks for the help and the program:)

**Comment by scwiterx** [comment]
![image](https://github.com/HarukaMa/starfield-xgp-import/assets/144282151/39b9e551-c0a3-46d9-915e-ef05ca0ceca1)

This is the save folder before converting. This save that is there is from the beginning that I made in the gamepass.

![image](https://github.com/HarukaMa/starfield-xgp-import/assets/144282151/e2123ea3-f4e4-4cd1-b605-3e2d3cf08dd8)
After the conversion.

![image](https://github.com/HarukaMa/starfield-xgp-import/assets/144282151/72ad53fa-6191-4d5a-97cb-7d73f025bc85)

**Comment by scwiterx** [comment]
OMG, IT WORKS. THANKS BRO!!

**Comment by mridul7x** [comment]
The dragging to cmd method works. Thanks!

**Comment by ParrishT3** [comment]
I am highly stupid when it comes to this. 

I open a CMD window in Admin mode, I drag main into it and not a thing happens at all. 

I open a CMD window no admin, it pops up with the path. I copy the code listed above and it gives me an "access is denied" error. 

I copied the path over to the Admin CMD and it still says Access Denied. What am I doing wrong? I think I'm missing an .exe file? All I have is what is in the download with the main.py and main.spec ? I do not see an sfsconverter.exe in the download or files section. 

EDIT: I double down on my stupid comment. I found the exe in releases (as it said) and it worked by dragging my .sfs into the converter no problem. Entered the game and it recognized it, no problem. 

It gave me a warning "This save file may rely on using content that is no longer there, blah blah blah "Constellation and Old Mars"

**Comment by Scooda** [comment]
I'm having the same issue @scwiterx had converter worked no problem. The game syncs with xbox for a few minutes before starting I'm assuming to read the new saves but they don't appear in game.
![image](https://github.com/HarukaMa/starfield-xgp-import/assets/104967145/98404635-f710-44a3-ac11-2c21cc44dce7)

**Comment by ParrishT3** [comment]
> I'm having the same issue @scwiterx had converter worked no problem. The game syncs with xbox for a few minutes before starting I'm assuming to read the new saves but they don't appear in game. ![image](https://user-images.githubusercontent.com/104967145/266196836-98404635-f710-44a3-ac11-2c21cc44dce7.png)

This is purely the cloud sync issue. I've seen issues on people saying it took up to 45 minutes to see their saves sync from one system to the other. I'd let it rest and pick it back up later just in case. Try relogging your game pass app and forcing a new sync?

**Comment by HarukaMa** [comment]
Updated the readme to add a note about the cloud sync issue.

It's definitely annoying as there is zero indicator if the sync is in progress. I had a check in the code to detect if there are state mismatches (usually caused by in-progress sync), but apparently it doesn't work every time.

**Comment by Amnessie** [comment]
@HarukaMa Hello First of all thank you for the great program! and also I have a question,  do you know if there is any way to edit the .sfs files if I open them now with Wordpad its all scrambled text and weird characters I`d like to compare a corrupt save file with a non corrupt save file to see what is the problem with the corrupt save files. 

Thank you!

**Comment by scwiterx** [comment]
> Estou tendo o mesmo problema@scwiterxo conversor funcionou sem problemas. O jogo sincroniza com o xbox por alguns minutos antes de começar. Presumo que vou ler os novos salvamentos, mas eles não aparecem no jogo. ![imagem](https://user-images.githubusercontent.com/104967145/266196836-98404635-f710-44a3-ac11-2c21cc44dce7.png)

I didn't do anything too much, I tested another save and it worked. But I believe that's what the friend said down there. You have to convert and wait a while for it to appear.

**Comment by HarukaMa** [comment]
> @HarukaMa Hello First of all thank you for the great program! and also I have a question, do you know if there is any way to edit the .sfs files if I open them now with Wordpad its all scrambled text and weird characters I`d like to compare a corrupt save file with a non corrupt save file to see what is the problem with the corrupt save files.
> 
> Thank you!

I briefly tried to see how the save file is constructed (besides the compressed chunks), but it seems pretty complicated. You can start from reading the save file using the code in [`savefile_types.py`](https://github.com/HarukaMa/starfield-xgp-import/blob/de7c041949db5372dacf5c3cc54808cdd5cd1d31/savefile_types.py#L15) to see if it can be at least read and decompressed.

**Comment by HarukaMa** [comment]
I'm converting this "issue" to discussion now as it is becoming too unfocused.

**Comment by zFlatliner** [comment]
Here's what I did to get it to work

Install Python
Download the source.zip from this Github
Extract files to C:\ (in my case I called it Starfieldxpg)
Run CMD
Change the directory to C:\Starfieldxpg
Type in main.py and then drag and drop your save game file onto the CMD window
Hit enter and it will convert it - make sure there is a space between main.py and the path to the saved game file
When you open the game you should see the saved file
When you load the game you might get an error loading Constellation and Old Mars - These are the premium add-ons - mostly skins so don't worry about them
Mods still seem to work as expected I am testing it now.

**Comment by bethannkushner** [comment]
does this work on saves converted to steam from XGP? trying to convert saves between XGP accounts. Ran in cmd and getting not supported error 0x103

**Reply by Amnessie** [reply]
Make sure to download the executable from releases and try it with that with the game closed and after putting the save back wait a bit for the cloud to upload that save

**Reply by HarukaMa** [reply]
Seems I might need to see if the new update changed the save file format, as 0x103 comes from save file parsing.
